### DISCUSSION TITLE
Whirring noise when not in use

### DISCUSSION THREAD

**Discussion by holycacao** [doudar/SmartSpin2k#473] [General]
Hi there,

I have a ~1 week old pre-built kit (thanks @eMadman !) that makes an odd whirring noise when not in use. I've attached an audio file where the first 3 seconds are just the baseline noise in the room for reference. Any ideas?

Firmware: `23.4.24-develop`
Bike: Bowflex C7

[SmartSpin2kNoise.zip](https://github.com/doudar/SmartSpin2k/files/12163069/SmartSpin2kNoise.zip)

**Comment by eMadman** [comment]
Hmm, this isn't one I've experienced when testing the assembled units.  Could you try unplugging it and reconnecting the power?  Does the issue persist?

If it does, can you try a firmware reset next?

**Reply by holycacao** [reply]
Just unplugged and reconnected. Will give it a few days and see if the noise starts happening again.

**Reply by holycacao** [reply]
I was waiting until I used the bike again to report back, but it will probably still be a couple of days. The noise is a bit quieter after the reboot, but still present. Could it be a fan? In the meantime, I just updated the firmware, so we'll see if that does the trick.

**Reply by doudar** [reply]
> I was waiting until I used the bike again to report back, but it will probably still be a couple of days. The noise is a bit quieter after the reboot, but still present. Could it be a fan? In the meantime, I just updated the firmware, so we'll see if that does the trick.

The only motor is the stepper motor which should power down whenever no app is connected. Are you able to rotate the knob freely when no app is connected? If there's resistance, that means the stepper driver isn't disabling properly or something is still accessing the SS2K.

It could also be a power issue. Either a bad filter capacitor on the main board (we haven't seen that yet on r3) or more likely a bad power supply. Do you have a different power supply that you can try? Anything between 9v-24v and >1a, as long as the barrel connector fits into the SS2K should be okay for testing. If you plan to use it long term, you'll want a >1.5A rating. 

I'd also try a different outlet. Possibly something else on your circuit is causing large power spikes.

**Reply by holycacao** [reply]
To update, the noise isn't really any better, even after the firmware update. And indeed, the stepper motor appears to be engaged - it is difficult to turn. If I unplug it, I can freely turn the knob. Are there any logs that would be helpful to further diagnose? Could the C7 be erroneously continuing to send a (bad) signal to the SS2K?

**Reply by holycacao** [reply]
Seeing some activity on this discussion. I will say that I don't hear the sound anymore and while I do turn off the device at night (because of the flashing lights), it does stay on during the day after use. I used FulGaz in the past, but I don't remember if I was using it when I originally posted this question. @NikkiSC09 said they ensured no apps were connected, but just thought I'd mention I was (probably) using FulGaz when I had this issue too.

**Reply by NikkiSC09** [reply]
Hi Holycacao,

Very interesting that you were using Fulgaz! I did not have the issue until I use that app today….

**Comment by NikkiSC09** [comment]
Hi,

I’ve had the device roughly a week and have started to hear the same whirring noise. The noise started today after connecting the device to FulGaz. I unplugged it restarted it and it went away. I just heard it again a few moments ago, and it scared the daylights out of me.

**Reply by eMadman** [reply]
Hi!  The immediate solution for now would be to disconnect the SmartSpin2k from power between rides.  The big connector on the breakout cable is there for easy disconnection of the whole device between rides.  

@doudar is on holiday in if there are firmware-related issues that require deeper investigation.  I'll get the ball rolling to collect the information we need to troubleshoot this:

1. Can you confirm the current version of the firmware that you are running?  You will see this in the SmartSpin2k Companion App.  See my screenshot below - this device is running 24.6.8-develop:
   ![Screenshot_20240623-140640](https://github.com/doudar/SmartSpin2k/assets/24726844/66a4c757-2fd4-4547-9e87-bcc7cb80cf6f)
2. Is the whirring sort of the sound you might get when there's active electronics?  I would describe it personally like a light buzzing sort of sound with no movement of the motor happening.  When the noise starts, is the knob is difficult to rotate by hand with the SmartSpin2k attached to the bike?  You might hear a clicking sound as you force the knob to rotate - this won't damage anything as long as you don't spin it by hand at high speed while it's plugged into power.  This tells me if the motor is powered and engaged when this happens.
3. Are there any training apps or devices running in the background that could be attempting to connect to the SmartSpin2k when this happens?
3. When this happens, can you shift up and down?  Does the motor turn as expected?
4. Can you launch your training app when this happens and get normal behavior on hills?
5. The last thing I'll ask you to do is to perform a firmware reset by clicking the "Set to Defaults" button in the SmartSpin2k Companion App.

![Screenshot_20240623-142725](https://github.com/doudar/SmartSpin2k/assets/24726844/80b7f8d5-7f09-40ea-8994-695816d4cacc)

**Reply by NikkiSC09** [reply]
Hi,

Thank you for replying!

Companion app the version I have running is 24.4.28. Your version is not showing as an option for me.

![IMG_0685](https://github.com/doudar/SmartSpin2k/assets/173628040/834ec9f0-3797-4839-b8f6-b8ac87d9dc4f)

The knob is not difficult at all to spend by hand. It did not seem that the motor was working in erg mode resistance was very tight and I could not pedal. The gears work fine when I switch to sim slope mode. The apps where I had the most problems with erg mode were in Rouvy. Fulgaz and Trainer Day were fine in sim modes.

The sound is loud and can be heard in another room when I’m not on the bike. I did disconnect from power and also from the wall outlet. I also ensured no other apps were connected to it.


I did a set default reset but still have no other versions than what I originally described at the beginning of the message.
